internal/artifactref/artifactref.go

```diff
@@ -0,0 +1,7 @@
+package artifactref
+
+import "regexp"
+
+const stepArtifactUsagePattern = `\$\(steps\.([^.]+)\.outputs(?:\.([^.]+))?\)`
```

Comment:
Just double checking: Its hardcoded to `outputs` only because we can only refer to `output` artifacts in following Steps right?

Reply:
Hmm. Because the artifacts just contain metadata, we should probably also allow passing `inputs`.  When a Step downloads an artifact, it reports the metadata of the downloaded artifacts in inputs. I think if a subsequent step needs to access the information, it should be possible to do that.

For example, when we create a Task-level artifact, users will probably want to fetch some step's artifacts (both inputs and outputs) and construct it.

Reply:
thanks just added inputs support